### PR TITLE
Cursor based pagination support for list groups api

### PR DESCRIPTION
**Pull request by symphony-soufiane**
### Description
Cursor based pagination support for apis:
- [x] list all groups
- [x] list all user following (already done)
- [x] list all user followers (already done)
- [x] list all user audit trail

### Dependencies
List the other pull requests that should be merged before/along this one.

### Checklist
- [ ] Referenced an issue in the PR title or description
- [x] Filled properly the description and dependencies, if any
- [x] Unit tests updated or added
- [x] Docstrings added or updated
- [x] Updated the documentation in [docsrc folder](../tree/main/docsrc)

### CODE REVIEW

File: symphony/bdk/core/service/user/user_service.py [hunk 1]
```diff
@@ -796,6 +796,40 @@ async def list_audit_trail(
...
+            start_timestamp: int,
+            end_timestamp: int = None,
+            initiator_id: int = None,
+            role: RoleId = None,
+            chunk_size: int = 100,
+            max_number: int = None
+    ) -> AsyncGenerator[V1AuditTrailInitiatorList, None]:
+        """Returns an asynchronous generation of audit trail of actions performed by a privileged user in a given period
+        of time.
+        See: `List Audit Trail v1 <https://developers.symphony.com/restapi/reference#list-audit-trail-v1>`_
```
Comment (symphony-elias) [inline symphony/bdk/core/service/user/user_service.py]: Please change the URL to:
`https://developers.symphony.com/restapi/reference/list-audit-trail-v1`

File: tests/resources/user/list_all_audit_trail.json [hunk 2]
```diff
@@ -0,0 +1,76 @@
+{
+  "items": [
+      {
+
```
Comment (symphony-elias) [inline tests/resources/user/list_all_audit_trail.json]: Please remove the extra line breaks

File: tests/ext/group_test.py [hunk 3]
```diff
@@ -117,6 +116,37 @@ async def test_list_groups(group_service, mocked_group, api_client):
...
+    args, kwargs = api_client.call_api.call_args
+
+    assert args[0] == '/v1/groups/type/{typeId}'
+    assert args[3] == [('limit', 2)]
+    assert len(groups) == 2
+    assert groups[0]['name'] == 'SDl test 0'
+    assert groups[1]['name'] == 'SDl test 1'
+
+@pytest.mark.asyncio
+async def test_list_all_groups_max_number(group_service, api_client):
```
Comment (symphony-elias) [inline tests/ext/group_test.py]: Could we test the case when we have two pages?

File: symphony/bdk/core/service/user/user_service.py [hunk 4]
```diff
@@ -796,14 +796,48 @@ async def list_audit_trail(
...
+    ) -> AsyncGenerator[V1AuditTrailInitiatorList, None]:
+        """Returns an asynchronous generation of audit trail of actions performed by a privileged user in a given period
+        of time.
+        See: `List Audit Trail v1 <https://developers.symphony.com/restapi/reference/list-audit-trail-v1>`_
+
+        :param start_timestamp: The start time of the period to retrieve the data.
+        :param end_timestamp:   The end time of the period to retrieve the data.
+        :param initiator_id:    The range and limit for pagination of data.
+        :param role:            Role to list audit trail for.
+        :param chunk_size:      The maximum number of audit trails to return. Default: 100.
```
Comment (symphony-elias) [inline symphony/bdk/core/service/user/user_service.py]: This is the maximum number of audit trails to return in one HTTP call

File: symphony/bdk/core/service/user/user_service.py [hunk 5]
```diff
@@ -796,14 +796,48 @@ async def list_audit_trail(
...
+        :param max_number:      The total maximum number of audit trails to retrieve.
```
Comment (symphony-elias) [inline symphony/bdk/core/service/user/user_service.py]: If set to None, we retrieve all audit trails until the last page

File: symphony/bdk/ext/group.py [hunk 6]
```diff
@@ -108,6 +111,30 @@ async def list_groups(self, status: Status = None, before: str = None, after: st
             kwargs["sort_order"] = sort_order
         return await self._group_api.list_groups(**kwargs)
 
+    @retry(retry=refresh_bearer_token_if_unauthorized)
```
Comment (symphony-elias) [inline symphony/bdk/ext/group.py]: `@retry` is not needed here because we already have it for the `list_groups` method

**Review by symphony-elias — APPROVED** [review]

File: symphony/bdk/ext/group.py (hunk 6 above)
Comment (symphony-soufiane) [reply symphony/bdk/ext/group.py]: Done in https://github.com/finos/symphony-bdk-python/pull/276/commits/a8d37738b59949adb32c2dd7dcf53eee730a2d28

**Review by symphony-elias — APPROVED** [review]
LGTM

**Pull request merged**